client/my-sites/plans/jetpack-plans/plan-upgrade/index.tsx

```diff
@@ -0,0 +1,128 @@
+import ProductCard from '../product-card';
+import ProductGridSection from '../product-grid/section';
+import { slugToSelectorProduct } from '../utils';
+import { getCurrentUserCurrencyCode } from 'calypso/state/current-user/selectors';
+import { getSelectedSiteId } from 'calypso/state/ui/selectors';
+
+/**
+ * Style dependencies
+ */
+import './style.scss';
+
+/**
+ * Type dependencies
+ */
+import type { PlanRecommendation } from './types';
+import type { Duration, PurchaseCallback, SelectorProduct } from '../types';
+
+type Props = {
+	planRecommendation: PlanRecommendation;
+	duration: Duration;
+	onSelectProduct: PurchaseCallback;
+};
+
+const PlanUpgradeSection: React.FC< Props > = ( {
+	planRecommendation,
+	duration,
+	onSelectProduct,
+} ) => {
+	const translate = useTranslate();
+	const siteId = useSelector( getSelectedSiteId );
+	const currencyCode = useSelector( getCurrentUserCurrencyCode );
+
+	let [ legacyPlan, newPlans ] = planRecommendation;
+
+	legacyPlan = getItemSlugByDuration( legacyPlan, duration );
+	newPlans = newPlans.map( ( p ) => getItemSlugByDuration( p, duration ) );
+
+	const legacyItem = slugToSelectorProduct( legacyPlan ) as SelectorProduct;
+	const newItems = newPlans.map( slugToSelectorProduct ) as SelectorProduct[];
+	const newItemCount = newItems.length;
+
+	return (
+		<ProductGridSection
+			title={ translate( 'We recommend {{product/}} based on your selection', {
+				components: {
+					product: (
+						<span className="plan-upgrade__product">
+							{ newItems.reduce( ( result, { displayName }, index ) => {
+								if ( result.length === 0 ) {
+									return [ displayName ];
+								}
+
+								if ( index === newItemCount - 1 ) {
+									return [ ...result, ' & ', displayName ];
+								}
+
+								return [ ...result, ', ', displayName ];
+							}, [] ) }
+						</span>
+					),
+				},
+			} ) }
+			className={ classNames( { 'with-single-reco': newItemCount === 1 } ) }
+		>
+			<p className="plan-upgrade__description">
+				{ translate(
+					"Jetpack has introduced new flexible plans and pricing options. As an added bonus, we're offering {{b}}%(percentage)s% off{{/b}} your first term.",
+					{
+						args: {
+							percentage: INTRO_PRICING_DISCOUNT_PERCENTAGE,
+						},
+						components: {
+							b: <strong />,
+						},
+					}
+				) }
+			</p>
+			<ul className="plan-upgrade__list">
+				<li className="plan-upgrade__legacy-item">
+					<ProductCard
+						item={ legacyItem }
+						siteId={ siteId }
+						currencyCode={ currencyCode }
+						selectedTerm={ duration }
+						hideSavingLabel
+						onClick={ onSelectProduct }
+					/>
+				</li>
+				<li className="plan-upgrade__separator">
+					<span className="plan-upgrade__arrow">{ '→' }</span>
+				</li>
+				<li className="plan-upgrade__new-items">
+					<ul className="plan-upgrade__new-items-list">
+						{ newItems.map( ( product ) => (
+							<li key={ product.iconSlug }>
```

Comment:
We are getting the typical React warning about not having unique keys. Maybe `iconSlug` is undefined or we need to use something else that is really unique.

![image](https://user-images.githubusercontent.com/3418513/116417793-98672280-a809-11eb-99fd-365d2416795f.png)

Reply:
Do you remember which values you used for the products, @rcanepa? I wasn't able to reproduce this.

Reply:
Yes, I used `?compare_plans=jetpack_premium,jetpack_security_daily,jetpack_backup_realtime`.

Reply:
This is really strange. It only happens to me with these specific products, and the debugger clearly shows they have different keys (at no point are they undefined) 🤔 .